### PR TITLE
Fixes improper escaping in edit message composer

### PR DESCRIPTION
**Pull request by yaya-usman**
Fixes:  [vector-im/element-web#22127](https://github.com/vector-im/element-web/issues/22127)

Signed-off-by: Yaya Usman [yaya-usman@users.noreply.github.com](mailto:yaya-usman@users.noreply.github.com) 

**Original Text**
original
**Before:**

onEditbefore

**After:** 
onEditAfter

Type: defect

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Fixes improper escaping in edit message composer ([\#8600](https://github.com/matrix-org/matrix-react-sdk/pull/8600)). Contributed by @yaya-usman.

### CODE REVIEW

File: src/editor/deserialize.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ const LIST_TYPES = ["UL", "OL", "LI"];
 
 // Escapes all markup in the given text
 function escape(text: string): string {
-    return text.replace(/[\\*_[\]`<]|^>/g, match => `\\${match}`);
+    return text.replace(/[\\*_[\]`<]|^>/g, match => `${match}`);
```
Comment (t3chguy) [inline src/editor/deserialize.ts:32]: This replace is now doing absolutely nothing? You're replacing `match` with itself

Comment (yaya-usman) [reply src/editor/deserialize.ts:32]: Yeah, you're right. I could have done better just wanted to push my changes to get assigned. So i can continue working on it later.

Comment (yaya-usman) [reply src/editor/deserialize.ts:32]: Nonetheless it seems to work without any problem even though it seems illogical, I'll probably need your help. I guess I need to figure out if any other part of the project is using this escape function if not I should probably remove it what do you think or figure out a better way to handle it?

Comment (t3chguy) [reply src/editor/deserialize.ts:32]: Well no, it regresses the reason this code was added in the first place.
See https://github.com/vector-im/element-web/issues/10722 + https://github.com/matrix-org/matrix-react-sdk/pull/8004

**Comment by t3chguy** [comment]
Abandoned - reopen if you are working on it.

**Pull request closed without merging**